src/components/inputs/select-input/select-input.js

```diff
@@ -70,14 +70,23 @@ export class SelectInput extends React.Component {
       option => (has(option, 'value') ? option : option.options)
     );
 
+    const selectedOptions = optionsWithoutGroups.filter(
+      option =>
+        props.isMulti
+          ? props.value.includes(option.value)
+          : has(option, 'value') && option.value === props.value
+    );
+
+    // We default to null, `type ValueType = OptionType | OptionsType | null | void`
+    /**
+     * we default to null
+     * > type ValueType = OptionType | OptionsType | null | void
+     * https://react-select.com/props#prop-types
+     */
     return {
       selectedOptions: props.isMulti
-        ? optionsWithoutGroups.filter(option =>
-            props.value.includes(option.value)
-          )
-        : optionsWithoutGroups.find(
-            option => has(option, 'value') && option.value === props.value
-          ),
+        ? selectedOptions
+        : selectedOptions[0] || null,
```

Comment:
I find this harder to read than it was before. I especially dislike that `selectedOptions` (the constant) doesn't actually hold the same value as `selectedOptions` (the state).

Could you use the approach we had when pairing?